autogen/agentchat/groupchat.py

```diff
@@ -74,6 +75,7 @@ def custom_speaker_selection_func(
     speaker_transitions_type: Literal["allowed", "disallowed", None] = None
     enable_clear_history: Optional[bool] = False
     send_introductions: bool = False
+    role_for_select_speaker_messages: Optional[str] = "system"
```

Comment:
If None case is not handled then the type hint should be just str or even better, Literal["user", "system", "assistant"]

Same with several other parameters in the class. We can have a separate PR to fix those parameters cc @joshkyh

Reply:
Sounds like a good idea to change the hint to Literal["user", "system", "assistant"] to assist developers.

I have a check for None or empty string in __post_init__:

```
        if self.role_for_select_speaker_messages is None or len(self.role_for_select_speaker_messages) == 0:
            raise ValueError("role_for_select_speaker_messages cannot be empty or None.")
```

Just confirming I'll leave this in? The hint doesn't enforce the value and it can still be None or empty.

Reply:
It's okay to do it in a next PR. I have created an issue.